docs/en/03_Upgrading/02_Upgrading_module.md

```diff
@@ -0,0 +1,331 @@
+      replacement: 'newBetterMethod'
+  props:
+    'ExampleUser\SilverstripeExampleModule\AmazingClass->oldProperty':
+      message: 'Replace with a different property'
+      replacement: 'newProperty'
+```
+
+When you are done updating you're `.upgrade.yml` file, you can call the `inspect` command to search for deprecated APIs.
+
+```bash
+# If upgrading from inside a test project
+upgrade-code inspect --root-dir vendor/example-user/silverstripe-example-module \
+  vendor/example-user/silverstripe-example-module/code/
+  
+# If upgrading the module directly
+upgrade-code inspect code/
+```
+
+## Step 6 - Update your entry point
+
+Module do not have an entry point. So there's nothing to do here.
+
+## Step 7 - Update project structure
+
+This step is optional. We recommend renaming `code` to `src`. This is only a convention and will not affect how your module will be executed.
+
+If you do rename this directory, do not forget to update your `autoload` configuration in your `composer.json` file.
+
+## Step 8 - Switch to public web-root
+
+The public web root does not directly affect module. So you can skip this step.
+
+## Step 9 - Move away from hardcoded paths for referencing static assets
+
+While SilverStripe 4 projects can get away with directly referencing static assets under some conditions, modules must dynamically exposed their static assets. This is necessary to move modules to the vendor folder and to enable the public web root.  
+
+### Exposing your module's static assets
+
+You'll need to update your module's `composer.json` file with a `extra.expose` key.
+
+
+```diff
+{
+    "name": "example-user/silverstripe-example-module",
+    "type": "silverstripe-vendormodule",
+    "require": {
+        "silverstripe/framework": "^4",
+        "silverstripe/vendor-plugin": "^1"
+    },
+    "autoload": {
+        "psr4": {
+            "ExampleUser\\SilverstripeExampleModule\\": "code/"
+        }
+    },
+    "autoload-dev": {
+        "psr4": {
+            "ExampleUser\\SilverstripeExampleModule\\Tests\\": "tests/"
+        }
+-    }
++    },
++    "extra": {
++        "expose": [
++            "images",
++            "styles",
++            "javascript"
++        ]
++    }
+}
+```
+
+### Referencing static assets
+
+This process is essentially the same for projects and modules. The only difference is that module static asset paths must be prefix with the module's name as defined in their `composer.json` file.
+
+```diff
+<?php 
+- Requirements::css('silverstripe-example-module/styles/admin.css');
++ Requirements::css('example-user/silverstripe-example-module: styles/admin.css');
+$pathToImage =
+-    'silverstripe-example-module/images/logo.png';
++    ModuleResourceLoader::singleton()->resolveURL('example-user/silverstripe-example-module: images/logo.png');
+```
+
+## Step 10 - Update database class references {#step10}
+
+Just like SilverStripe projects, your module must define class names remapping for every DataObject child.
```

Comment:
```suggestion
Just like projects, your module must define class names remapping for every DataObject child.
```

Only because we never prefix with "SilverStripe" in the rest of this doc.